server/consumer.go

```diff
@@ -3374,6 +3374,25 @@ func (o *consumer) isFilteredMatch(subj string) bool {
 	return false
 }
 
+// Check if the candidate filter subject is equal to or a subset match
```

Comment:
Candidate is not a filter subject per se, and needs to be a literal.

Reply:
Oh, does it? Maybe I didn't fully understand how this code is being used then. I was assuming that it gets called when `nats stream purge --subject=SUBJECT` is used. And I think I have successfully used this command with subjects that have a wildcard.